language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/documentsymbol/DocumentSymbolResolver.java

```diff
@@ -151,20 +177,146 @@
+    @Override
+    public Optional<DocumentSymbol> transform(EnumDeclarationNode enumDeclarationNode) {
+        String name = enumDeclarationNode.identifier().text();
+        SymbolKind symbolKind = SymbolKind.Enum;
+        Range range = DocumentSymbolUtil.generateNodeRange(enumDeclarationNode);
+        Optional<MetadataNode> metadata = enumDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ModuleXMLNamespaceDeclarationNode moduleXMLNamespaceDeclarationNode) {
+        String name = SyntaxKind.XMLNS_KEYWORD.stringValue() + " "
+                + moduleXMLNamespaceDeclarationNode.namespaceuri().toSourceCode();
+        Optional<IdentifierToken> prefix = moduleXMLNamespaceDeclarationNode.namespacePrefix();
+        if (prefix.isPresent()) {
+            name = name + " as " + prefix.get();
+        }
+        SymbolKind symbolKind = SymbolKind.Namespace;
+        Range range = DocumentSymbolUtil.generateNodeRange(moduleXMLNamespaceDeclarationNode);
+        return Optional.ofNullable(createDocumentSymbol(name, symbolKind, null,
+                range, range, false, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(ListenerDeclarationNode listenerDeclarationNode) {
+        StringBuilder name = new StringBuilder(SyntaxKind.LISTENER_KEYWORD.stringValue());
+        Optional<TypeDescriptorNode> typeDescriptorNode = listenerDeclarationNode.typeDescriptor();
+        if (typeDescriptorNode.isPresent()) {
+            name.append(" " + typeDescriptorNode.get().toSourceCode());
+        }
+        name.append(" " + listenerDeclarationNode.variableName().text());
+        SymbolKind symbolKind = SymbolKind.Object;
+        Range range = DocumentSymbolUtil.generateNodeRange(listenerDeclarationNode);
+        Optional<MetadataNode> metadata = listenerDeclarationNode.metadata();
+        boolean isDeprecated = metadata.isPresent() &&
+                DocumentSymbolUtil.isDeprecated(metadata.get());
+        return Optional.ofNullable(createDocumentSymbol(name.toString(), symbolKind, null,
+                range, range, isDeprecated, Collections.emptyList(), this.context));
+    }
+
+    @Override
+    public Optional<DocumentSymbol> transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        StringBuilder name = new StringBuilder(SyntaxKind.ANNOTATION_KEYWORD.stringValue());
+        Optional<Node> typeDesc = annotationDeclarationNode.typeDescriptor();
+        if (typeDesc.isPresent()) {
+            name.append(" ").append(typeDesc.get().toSourceCode());
+        }
+        name.append(" ").append(annotationDeclarationNode.annotationTag().text());
```

Comment:
If we have a name for the symbol, let's stick to the name itself without modifying it. Otherwise we can generate the name value with the meta information. Please check this in other places as well

Reply:
Your suggestion is to remove `annotation` keywords and the rest right? Will do

Reply:
Yes

Reply:
Done